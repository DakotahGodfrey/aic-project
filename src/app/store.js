import { configureStore } from '@reduxjs/toolkit';
import homeReducer from 'app/slices/homeSlice';
import galleryReducer from 'app/slices/gallerySlice';
import artistsReducer from 'app/slices/artistsSlice';
import artworksReducer from 'app/slices/artworkSlice';
import artistReducer from 'app/slices/artistSlice';

export const store = configureStore({
  reducer: {
    home: homeReducer,
    gallery: galleryReducer,
    artists: artistsReducer,
    artwork: artworksReducer,
    artist: artistReducer,
  },
});
