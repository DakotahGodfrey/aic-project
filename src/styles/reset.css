*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
}
html {
  font-size: 10px;
}
ol,
ul {
  list-style: none;
}
a {
  text-decoration: none;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
}
::-webkit-scrollbar {
  width: 0; /* Remove scrollbar space */
  background: transparent; /* Optional: just make scrollbar invisible */
}
