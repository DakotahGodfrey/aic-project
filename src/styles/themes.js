const defaultTheme = {
  textColor: '#222',
  secondaryColor: '#fff',
  navBG: '#111',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#666',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};
const darkTheme = {
  textColor: '#eee',
  navBG: '#111',
  secondaryColor: '#151515',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#ddd',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};

const getFromTheme = (propName) => (props) => props.theme[propName];
export { defaultTheme, darkTheme, getFromTheme };
