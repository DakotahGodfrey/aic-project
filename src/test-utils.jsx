import * as React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

function render(ui, { theme, store, ...options } = {}) {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <Router>{children}</Router>
      </ThemeProvider>
    </Provider>
  );
  return rtlRender(ui, { wrapper: Wrapper, ...options });
}

export * from '@testing-library/react';
// override React Testing Library's render with our own
export { render };
