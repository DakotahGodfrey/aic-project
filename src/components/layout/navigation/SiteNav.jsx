import React, { useState, useEffect } from 'react';

import styled from 'styled-components/macro';
import { Link } from 'react-router-dom';
import { getFromTheme } from 'styles/themes';
import useViewport from 'utils/hooks/useViewport';
import { mediaQueries } from 'utils/data';
import { slideInFromRight } from 'styles/animations/keyframes';
import MenuButton from '../buttons/MenuButtton';
import ThemeToggle from '../buttons/ThemeToggle';

const PageNav = styled.nav`
  position: sticky;
  top: 0;
  padding: 2rem 3rem;
  background: ${getFromTheme('navBG')};
  z-index: 2;
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 13rem;
  box-shadow: 0 2px 4px ${getFromTheme('secondaryColor')};
  ul {
    display: flex;
    justify-content: space-between;
    align-items: center;
    font-size: 1.8rem;
    li:not(:first-child) {
      margin-left: 1.4rem;
    }
    a {
      display: block;
      padding: 0.75rem 1.5rem;
      color: ${getFromTheme('lightColor')};
      border-radius: 1rem;
      transition: background-color 0.4s ease-out;
      &:hover,
      &:focus {
        background-color: ${getFromTheme('brandColor')};
      }
    }
  }
`;

const Logo = styled.div`
  height: 4.4rem;
  width: 4.4rem;
  font-size: 1.4rem;
  a {
    transition: all 0.4s ease-out;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 100%;
    width: 100%;
    color: ${getFromTheme('lightColor')};
    padding: 0.75rem 0.75rem;
    border-radius: 50%;
    span {
      font-size: 3.4rem;
    }
  }
  a:hover,
  a:focus {
    background-color: ${getFromTheme('brandColor')};
    border-color: transparent;
  }
`;

const MobileMenu = styled.ul.attrs((props) => ({
  id: 'navMenu',
}))`
  position: fixed;
  top: 13rem;
  display: flex;
  width: 100%;
  flex-direction: column;
  align-items: center;
  font-size: 1.8rem;
  text-align: center;
  padding: 1rem 0;
  background: rgba(17, 17, 17, 0.9);
  z-index: 3;
  animation: ${slideInFromRight} 1s ease-out forwards;
  li {
    width: 100%;
  }
  a {
    text-transform: capitalize;
    color: ${getFromTheme('lightColor')};
    display: flex;
    width: 100%;
    min-height: 4.4rem;
    align-items: center;
    justify-content: center;
    padding: 0.75rem 1.5rem;
    border-radius: 1rem;
    transition: background-color 0.4s ease-out;
    &:hover {
      color: ${getFromTheme('lightColor')};
      background-color: ${getFromTheme('brandColor')};
    }
  }
`;

const ButtonContainer = styled.div`
  display: flex;
  button:not(:first-child) {
    margin-left: 1.6rem;
  }
`;

const SiteNav = ({ links, toggleTheme, theme }) => {
  const [isActive, setIsActive] = useState(false);

  const handleMenuClick = () => {
    setIsActive(!isActive);
  };
  const handleHomeClick = () => {
    if (isActive) {
      setIsActive(false);
    }
  };
  const { width } = useViewport();
  const { small } = mediaQueries;
  const isSmall = width < small;
  useEffect(() => {
    if (isActive) {
      document.getElementById('1').focus();
    }
  }, [isActive]);
  return (
    <>
      <PageNav>
        <Logo>
          <Link onClick={() => handleHomeClick()} to={'/'} id='home'>
            <span className='material-icons'>home</span>
          </Link>
        </Logo>
        {isSmall ? (
          <ButtonContainer>
            <MenuButton
              isActive={isActive}
              handleClick={() => handleMenuClick()}
              children={isActive ? 'close' : 'menu'}
            />
            <ThemeToggle
              handleClick={() => toggleTheme()}
              isPressed={theme !== 'default'}
            >
              <i className='material-icons'>
                {theme === 'default' ? 'mode_night' : 'light_mode'}
              </i>
            </ThemeToggle>
          </ButtonContainer>
        ) : (
          <ul>
            {links.map((link) => (
              <li key={link.id}>
                <Link to={link.path} children={link.display} />
              </li>
            ))}
            <li>
              <ThemeToggle
                handleClick={() => toggleTheme()}
                isPressed={theme !== 'default'}
                aria-label='toggle theme'
              >
                <i className='material-icons'>
                  {theme === 'default' ? 'mode_night' : 'light_mode'}
                </i>
              </ThemeToggle>
            </li>
          </ul>
        )}
      </PageNav>
      {isActive && isSmall && (
        <MobileMenu>
          {links.map((link) => (
            <li key={link.id}>
              <Link
                onClick={handleMenuClick}
                to={link.path}
                id={link.id}
                children={link.display}
              />
            </li>
          ))}
        </MobileMenu>
      )}
    </>
  );
};

export default SiteNav;
