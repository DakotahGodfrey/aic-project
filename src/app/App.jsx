import React, { useEffect, useState } from 'react';
import { ThemeProvider } from 'styled-components';
import { darkTheme, defaultTheme } from 'styles/themes';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  useLocation,
} from 'react-router-dom';
import SiteNav from 'components/layout/navigation/SiteNav';
import { siteLinks } from 'utils/data';
import Home from 'components/screens/Home';
import Gallery from 'components/screens/Gallery';
import Artists from 'components/screens/Artists';
import Artwork from 'components/screens/Artwork';
import Artist from 'components/screens/Artist';
import AboutApp from 'components/screens/AboutApp';
import NotFound from 'components/screens/NotFound';
import ScrollUpButton from 'components/layout/buttons/ScrollUpButton';

function ScrollToTop() {
  const { pathname } = useLocation();

  useEffect(() => {
    if (window instanceof Window) {
      window.scrollTo(0, 0);
    }
  }, [pathname]);

  return null;
}

const App = () => {
  const [theme, setTheme] = useState('default');

  const handleThemeToggle = () => {
    if (theme === 'default') {
      setTheme('darkTheme');
    } else {
      setTheme('default');
    }
    console.log(theme);
  };
  return (
    <ThemeProvider theme={theme === 'default' ? defaultTheme : darkTheme}>
      <Router>
        <div
          id='app'
          style={
            theme === 'default'
              ? { backgroundColor: '#eee' }
              : { backgroundColor: '#111' }
          }
        >
          <ScrollToTop />
          <SiteNav
            links={siteLinks}
            toggleTheme={handleThemeToggle}
            theme={theme}
          />
          <Switch>
            <Route exact path={'/'} component={Home} />
            <Route exact path={'/gallery'} component={Gallery} />
            <Route exact path={'/artists'} component={Artists} />
            <Route exact path={'/gallery/artwork/:id'} component={Artwork} />
            <Route exact path={'/artists/:id'} component={Artist} />
            <Route exact path={'/about'} component={AboutApp} />
            <Route component={NotFound} />
          </Switch>
          <ScrollUpButton />
        </div>
      </Router>
    </ThemeProvider>
  );
};

export default App;
