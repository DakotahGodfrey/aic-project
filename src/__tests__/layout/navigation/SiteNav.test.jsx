import { store } from 'app/store';
import SiteNav from 'components/layout/navigation/SiteNav';
import { defaultTheme } from 'styles/themes';
import { screen, render, fireEvent } from 'test-utils';

const siteLinks = [
  { id: 1, path: '/test', display: 'Test' },
  { id: 2, path: '/test2', display: 'Test2' },
  { id: 3, path: '/test3', display: 'Test3' },
];

let theme = 'default';
const toggleTheme = jest.fn(() => {
  theme = theme === 'default' ? 'dark' : 'default';
  console.log(theme);
});

describe('<SiteNav />', () => {
  it.skip('throws an error if an array of links are not provided', () => {
    expect(() => {
      render(<SiteNav />);
    }).toThrow();
  });
  it('matches the snapshot', () => {
    const { container } = render(<SiteNav links={siteLinks} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders an accessible navigation element', () => {
    render(<SiteNav links={siteLinks} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByRole('navigation')).toBeTruthy();
  });
  it('renders a link to the home page', () => {
    render(<SiteNav links={siteLinks} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByRole('link', { name: /home/i })).toHaveAttribute(
      'href',
      '/'
    );
  });
  it('renders a link for each item in the links prop with the correct text content', () => {
    render(<SiteNav links={siteLinks} />, {
      theme: defaultTheme,
      store: store,
    });
    siteLinks.forEach((link) => {
      expect(screen.getByRole('link', { name: link.display })).toHaveAttribute(
        'href',
        link.path
      );
    });
  });

  it('renders a menu button if the screen size is smaller thanthe breakpoint', () => {
    window.innerWidth = 500;
    const { rerender } = render(<SiteNav links={siteLinks} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByRole('button', { name: /menu/i })).toBeInTheDocument();
    window.innerWidth = 1000;
    rerender();
    expect(
      screen.queryByRole('button', { name: /menu/i })
    ).not.toBeInTheDocument();
  });

  it('renders a list of links when the menu button is clicked and screen  size is small', () => {
    window.innerWidth = 500;
    render(<SiteNav links={siteLinks} />, {
      theme: defaultTheme,
      store: store,
    });

    const menuButton = screen.getByRole('button', { name: /menu/i });
    expect(screen.queryAllByRole('link')).toHaveLength(1);
    fireEvent.click(menuButton);
    expect(screen.getAllByRole('link')).toHaveLength(siteLinks.length + 1);
  });
  it('renders a button with the text set to close when the menu button is clicked', () => {
    window.innerWidth = 500;
    render(<SiteNav links={siteLinks} />, {
      theme: defaultTheme,
      store: store,
    });
    const menuButton = screen.getByRole('button', { name: /menu/i });
    expect(menuButton).toBeTruthy();
    fireEvent.click(menuButton);
    expect(screen.getByRole('button', { name: /close/i })).toBeTruthy();
    fireEvent.click(menuButton);
    expect(screen.getByRole('button', { name: /menu/i })).toBeTruthy();
  });

  it('renders a themeToggle Button', () => {
    const { rerender } = render(
      <SiteNav links={siteLinks} theme={theme} toggleTheme={toggleTheme} />,
      {
        theme: defaultTheme,
        store: store,
      }
    );
    const themeToggle = screen.getByRole('button', { name: /mode_night/i });
    expect(themeToggle).toBeTruthy();
    fireEvent.click(themeToggle);
    expect(toggleTheme).toHaveBeenCalled();
    theme = 'dark';
    rerender(
      <SiteNav links={siteLinks} theme={theme} toggleTheme={toggleTheme} />,
      {
        theme: defaultTheme,
        store: store,
      }
    );
    expect(screen.getByRole('button', { name: /light_mode/i })).toBeTruthy();
  });
});
