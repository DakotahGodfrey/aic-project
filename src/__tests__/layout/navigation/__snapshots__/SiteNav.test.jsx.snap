// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteNav /> matches the snapshot 1`] = `
<nav
  class="SiteNav__PageNav-sc-gqtbjf-0 gfyDmB"
>
  <div
    class="SiteNav__Logo-sc-gqtbjf-1 bdklag"
  >
    <a
      href="/"
      id="home"
    >
      <span
        class="material-icons"
      >
        home
      </span>
    </a>
  </div>
  <ul>
    <li>
      <a
        href="/test"
      >
        Test
      </a>
    </li>
    <li>
      <a
        href="/test2"
      >
        Test2
      </a>
    </li>
    <li>
      <a
        href="/test3"
      >
        Test3
      </a>
    </li>
    <li>
      <button
        aria-pressed="true"
        class="sc-gtsrHT bsVBCn"
      >
        <i
          class="material-icons"
        >
          light_mode
        </i>
      </button>
    </li>
  </ul>
</nav>
`;
